### ISSUE TITLE
Regarding 3_segmentation pretrained models

### ISSUE THREAD

**Issue by jnikhilreddy** [Attila94/CIConv#3]
Hi,

thanks for posting the code and models. I was trying out the 3_segmentation code folder. I was testing with refinenet and W-refinenet on night time driving dataset. I was getting 33.5 and 25 mIOU scores respectively. Can you let me know what could be a possible reason? Are the results reported on 1024*512 resolution images?

**Comment by Attila94** [comment]
Hi Nikhil,

Thanks for your interest and thanks for reporting this. I have re-run the evaluation script for the segmentation checkpoints and there indeed seem to be some inconsistencies with the results reported in the paper. This has two causes:

1. When running the experiments I came across some labeling errors in the NighttimeDriving test set. After emailing the authors of the dataset, I have decided to manually correct these myself. I will make a note of this in the readme and upload my corrected labels to the repository. To be precise, I have corrected the following labels:  
    - incorrectly labeled car on right in `0_frame_0205_gtCoarse_color.png`
    - incorrectly labeled building in center of `0_frame_0273_gtCoarse_color.png`
    - incorrectly labeled building in center of `0_frame_0304_gtCoarse_color.png`
    - incorrectly labeled person on left in `0_frame_0461_gtCoarse_color.png`
    - incorrectly labeled pole on left in `0_frame_0604_gtCoarse_color.png`
    - missing bicycle on right in `0_frame_1120_gtCoarse_color.png`
    - corrupted label center-left part of `0_frame_2548_gtCoarse_color.png`
    - incorrectly labeled poles and traffic lights in `01_frame_0555_gtCoarse_color.png`
    - incorrectly labeled pole on right in `01_frame_0744_gtCoarse_color.png`
    - incorrectly labeled car in `01_frame_0906_gtCoarse_color.png`
    - incorrectly labeled cars in `01_frame_0956_gtCoarse_color.png`
    - incorrectly labeled road in `01_frame_1195_gtCoarse_color.png`
    - small corruption near car on left in `01_frame_1285_gtCoarse_color.png`

2. I mistakenly seem to have copied the mIoU of the second-to-last epoch instead of the final score for the baseline (`cityscapes_rgb.pth.tar`), i.e. 34.1 mIoU instead of 33.2 mIoU.

To summarize:

|                      | Reported in paper | Original (incorrect) labels | Corrected labels |
| -------------------- | ----------------- | --------------------------- | --------- |
| RefinNet [baseline]  | 34.1              | 33.5                        | **33.2**      |
| *W*-RefineNet [ours] | 41.6              | 41.2                        | **41.6**      |

When running the evaluation script for *W*, make sure to pass `--invariant 'W'` as an argument, i.e. `python eval.py 'cityscapes_w.pth.tar' --invariant 'W'`. On the original NighttimeDriving dataset (with incorrect labels) this should give 

```   
---------------------
Epoch 0
classes           IoU
---------------------
road          : 0.889
sidewalk      : 0.526
building      : 0.819
wall          : 0.161
fence         : 0.000
pole          : 0.382
traffic light : 0.507
traffic sign  : 0.606
vegetation    : 0.420
terrain       : 0.000
sky           : 0.385
person        : 0.502
rider         : 0.284
car           : 0.626
truck         : 0.000
bus           : 0.732
train         : 0.657
motorcycle    : 0.000
bicycle       : 0.323
---------------------
Score Average : 0.412
---------------------
```

The input resolution is indeed 1024*512.

Please let me know if this helps!

**Comment by hassaanmahmood** [comment]
@Attila94 using  `python eval.py '/CIConv-main/experiments/3_segmentation/cityscapes_w.pth.tar' --invariant 'W'`
I am getting following scores. I am getting 0.216 for Dark Zurich .

Night time driving
Score Average : 0.412
But for DarkZurich
Score Average : 0.216
Although I am getting this warning, but I do not think that can be the issue
/lib/python3.7/site-packages/torchvision/transforms/functional.py:405: UserWarning: Argument interpolation should be of type InterpolationMode instead of int. Please, use InterpolationMode enum.
  "Argument interpolation should be of type InterpolationMode instead of int. "
/lib/python3.7/site-packages/torchvision/transforms/functional.py:405: UserWarning: Argument interpolation should be of type InterpolationMode instead of int. Please, use InterpolationMode enum.
  "Argument interpolation should be of type InterpolationMode instead of int. "
/lib/python3.7/site-packages/torchvision/transforms/functional.py:405: UserWarning: Argument interpolation should be of type InterpolationMode instead of int. Please, use InterpolationMode enum.
  "Argument interpolation should be of type InterpolationMode instead of int. "
/lib/python3.7/site-packages/torchvision/transforms/functional.py:405: UserWarning: Argument interpolation should be of type InterpolationMode instead of int. Please, use InterpolationMode enum.
  "Argument interpolation should be of type InterpolationMode instead of int. "
